### ISSUE TITLE
Improve stack trace for `await for(var item in someList) {` 

### ISSUE THREAD

**Issue by zoechi** [dart-lang/sdk#24084]
The stack trace produced by 
```dart
main() {
  someAsync();
}

someAsync() async {
  var list = ['a', 'b', 'c'];
  await for(var l in list) {
    print(l);
  }
}
```
doesn't contain the line faulty line

```
Unhandled exception:
Uncaught Error: type 'List' is not a subtype of type 'Stream' of 'stream'.
Stack Trace:
#0      StreamIterator.StreamIterator (dart:async/stream.dart:1676)
#1      someAsync.<someAsync_async_body> (file:///home/zoechi/dart/playground/server/bin/bug_report/await_for_list/main.dart)
#2      Future.Future.microtask.<anonymous closure> (dart:async/future.dart:144)
#3      _microtaskLoop (dart:async/schedule_microtask.dart:43)
#4      _microtaskLoopEntry (dart:async/schedule_microtask.dart:52)
#5      _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:96)
#6      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:149)

#0      _rootHandleUncaughtError.<anonymous closure> (dart:async/zone.dart:895)
#1      _microtaskLoop (dart:async/schedule_microtask.dart:43)
#2      _microtaskLoopEntry (dart:async/schedule_microtask.dart:52)
#3      _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:96)
#4      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:149)

Process finished with exit code 255
```

**Comment by fsc8000** [comment]
Proposed fix: https://codereview.chromium.org/1516493002/

The stack trace contains the missing source position:
```
Unhandled exception:
type 'List' is not a subtype of type 'Stream' of 'stream'.
#0      StreamIterator.StreamIterator (dart:async/stream.dart:1684)
#1      someAsync.<someAsync_async_body> (file:///media/SSD/dartgit2/sdk/st.dart:7:22)
#2      Future.Future.microtask.<anonymous closure> (dart:async/future.dart:144)
#3      _microtaskLoop (dart:async/schedule_microtask.dart:41)
#4      _startMicrotaskLoop (dart:async/schedule_microtask.dart:50)
#5      _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:96)
#6      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:149)
```

**Comment by fsc8000** [comment]
Fixed in 12155b1920b19c0f6d743a841cbe3c36145f6046.
